base/uk.ac.stfc.isis.ibex.ui.synoptic.editor/src/uk/ac/stfc/isis/ibex/ui/synoptic/editor/target/selector/TargetSelectorPanel.java

```diff
@@ -0,0 +1,234 @@
+
+        viewModel.addPropertyChangeListener("enabled", new PropertyChangeListener() {
+            @Override
+            public void propertyChange(PropertyChangeEvent evt) {
+                Utils.recursiveSetEnabled(TargetSelectorPanel.this, viewModel.isEnabled()); 
+            }
+        });
+        
+        // Have to do this here otherwise the binding overrides the default value in the model.
+        Utils.recursiveSetEnabled(this, viewModel.isEnabled());
+        
+        bind();
+    }
+    
+    private void drawNameControl() {
+        Composite container = new Composite(this, SWT.NONE);
+        container.setLayout(new GridLayout(2, false));
+        container.setLayoutData(new GridData(SWT.FILL, SWT.TOP, true, false));
+
+        Label lblName = new Label(container, SWT.NONE);
+        lblName.setLayoutData(new GridData(SWT.LEFT, SWT.CENTER, false, false));
+        lblName.setText("Name: ");
+
+        txtName = new Text(container, SWT.BORDER);
+        txtName.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
+        txtName.addVerifyListener(new ComponentNameVerifier());
+    }
+
+    @SuppressWarnings("checkstyle:magicnumber") // Gui method   
+    private void drawControls() {
+        Composite container = new Composite(this, SWT.NONE);
+        container.setLayout(new GridLayout(3, false));
+        container.setLayoutData(new GridData(SWT.FILL, SWT.TOP, true, false));
+        
+        Label lblIcon = new Label(container, SWT.NONE);
+        lblIcon.setLayoutData(new GridData(SWT.LEFT, SWT.CENTER, false, false));
+        lblIcon.setText("Icon: ");
+
+        comboType = new ComboViewer(container);
+        comboType.getCombo().setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false, 2, 1));
+        comboType.getCombo().setItems(viewModel.componentTypesArray);
+        
+        Label lblSelectedTarget = new Label(container, SWT.NONE);
+        lblSelectedTarget.setLayoutData(new GridData(SWT.LEFT, SWT.CENTER, false, false));
+        lblSelectedTarget.setText("Target OPI: ");
+
+        txtSelectedTarget = new Text(container, SWT.BORDER | SWT.READ_ONLY);
+        txtSelectedTarget.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
+        
+        Button buttonClearTarget = new Button(container, SWT.NONE);
+        buttonClearTarget.setText("Clear");
+        buttonClearTarget.addSelectionListener(new SelectionListener() {         
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+                viewModel.clearTarget();
+            }
+            
+            @Override
+            public void widgetDefaultSelected(SelectionEvent e) {
+                widgetSelected(e);
+            }
+        });
+    }
+
+    @SuppressWarnings("checkstyle:magicnumber") // Gui method
+    private void drawTargetTree() {
+
+        final List<String> keyset = new ArrayList<>(new TreeSet<>(availableOpis.keySet()));
+        Collections.sort(keyset);
```

Comment:
I think this is redundant, it's only sorting a list with one element

Reply:
I'm confused by this comment, the list definitely has more than one element. Am I missing something here?

```java
        Collections.sort(keyset);

        System.out.println("elements in the list:");
        for (String key : keyset) {
            System.out.print(key + ", ");
        }
        System.out.println("");
```

```
elements in the list:
All devices, Choppers, Cryogenics, Gas handling systems, Imaging cameras, Jaws and slits, Loading rigs, Magnets, Miscellaneous motion control, Monitors, Multimeters, Power supplies, Pressure sensors, Sample changers, Sample stacks and goniometers, Temperature controllers, Uncategorised devices, Water baths, 
```

Reply:
Sorry, you're right, that's what I get for reviewing code at 8pm!